src/Files.App/Helpers/RecycleBinHelpers.cs

```diff
@@ -76,6 +80,7 @@ public async Task EmptyRecycleBin()
             if (result == ContentDialogResult.Primary)
             {
                 Shell32.SHEmptyRecycleBin(IntPtr.Zero, null, Shell32.SHERB.SHERB_NOCONFIRMATION | Shell32.SHERB.SHERB_NOPROGRESSUI);
+                RaiseRecycleBinChangedEvent();
```

Comment:
You can use null check
```csharp
RecycleBinChanged?.Invoke(null, null);
```

Reply:
Changed inside the function that handles the event raising. But would you rather have the event raised directly without any method maybe?

Reply:
Yes. There's no need for additional method to raise this event

Reply:
We actuallty do need it to raise the event outside of the class. We need to raise it from `RestoreItemsFromTrashAsync` and `DeleteItemsAsync` from `FilesystemHelpers.cs`.

Reply:
@d2dyno1 Should I do something to indeed remove the need of the method entirely or should it stay that way? 

Removing it would means to move methods from the `FileSystemHelpers` to the `RecycleBinHelpers`, albeit it wouldn't be their optimal locations.

Reply:
I think moving it would make sense (?). In addition, it should be moved to instantiable helper (because it exposes an event) but that'd be part of a bigger refactor out of scope for this PR.

Reply:
I think that moving it is better and good

Reply:
Not sure how to actually implement this. I have been turning the issue in my head for the past few days, but I cannot find a solution that feels actually clean for it. _Consider this message as a formal request for help._

After giving it some thoughts, the methods from `FilesystemHelpers.cs` should **not** be moved. They are in their right places, they are not only linked to the recycle bin. 

A possibility would to raise events in `FilesystemHelpers.cs` when `Delete` or `Restore` are called, and have methods to update the recycle bin icon subscribe to them. But then, an issue comes from the **RecycleBinHelpers.EmptyRecycleBin()** method that does not use the `Delete` method to clear the recycle bin.

I must be missing something obvious, and I'd like to see this PR through.